aggregate.js

```diff
@@ -118,7 +122,7 @@ export const ReactiveAggregate = (sub, collection = null, pipeline = [], options
   try { _CircDepPreventionSimpleSchema = require('simpl-schema'); } catch (e) { packageErrors.push({ name: 'simpl-schema', error: e }); }
   const isUsingMongoObjectIDSupport = packageErrors.length === 0;
   if (!isUsingMongoObjectIDSupport && !_errorsDisplayedOnce) {
-    if (localOptions.warnings) {
+    if (localOptions.warnings && localOptions.objectIdWarnings) {
```

Comment:
Oh. I should perhaps add that I've not fully worked out how this would work in code, but I'm thinking string -> string interpolation via template literal.

```js
const w_map = { 'w_deprecations': 1, 'w_object_id': 2, ... };
...
'!w_object_id' -> `${!w_map['w_object_id']}`
```

or something like that.

Reply:
Yes it also crossed my mind multiple times to accommodate other future warnings.
I think silencing deprecation warnings is just as risky as silencing all warnings. So that didn't motivate me much to go into future proofing.

If it was my own code I would have definitely done something future-proof but because it's your code and presumably your style I was avoiding being adventurous, just doing the minimum code within the repo's style to get rid of the unwanted warnings.

But now that you've shown your support for the future proofing approach and are open to discussing it, I'm happy to add to add my suggestion.

I dislike the bitmap idea. (I mean I'll be happy with anything that can target ObjectId, but since we're discussing it...) I think that bitmaps are bit silly in JavaScript which is such an inefficient language anyway, it's not like there will be any performance gains in doing it and it will likely alienate users who aren't familiar with computer science fundamentals (binary number systems, logical AND OR etc) for no measurable gain.

So while I'm happy with whatever approach you go with, I'll show you the future-proof idea that I had, and you can see if you like it or not etc.

Reply:
@robfallows here's my suggestion for a future proof approach.
https://github.com/softwarecreations/tunguska-reactive-aggregate/commit/7013aef9cec636f827d6e35d3bc6e4472405171e

I didn't update the readme etc. See what you think. If you want to do something different, then please go ahead. I'm not particular about how it gets done, as long as it gets done :)

Reply:
In terms of warnings:false killing (and overriding specific options) all warnings. That's pretty standard in most software. If people specify a specific warning, they're only going to specify the ones that they want to silence anyway. I think the bitmap thing is way over-complicated for what is needed. But you do you, I really don't mind any approach. I just gave my opinion because you asked.

Reply:
@robfallows great! I've updated the README.md and History.md accordingly.

How and when do you want to proceed with merging it?